### PR TITLE
Kruize api update result

### PR DESCRIPTION
**Pull request by msvinaykumar**
Changes related to Update metric results API.

### CODE REVIEW

File: src/main/java/com/autotune/common/k8sObjects/ExperimentResultData.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import com.autotune.utils.AnalyzerConstants;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Experiment result storage object.
+ */
+public class ExperimentResultData {
```
Comment (dinogun) [inline src/main/java/com/autotune/common/k8sObjects/ExperimentResultData.java:26]: In general we need to have classes inside `k8sObjects` dir if it's associated with a CRD or is directly related to k8s

File: src/main/java/com/autotune/analyzer/Experimentator.java [hunk 2]
```diff
@@ -20,125 +20,124 @@
...
+    @Override
+    public void run() {
+
+    }
+
+    /**
+     * @param kruizeObject
+     * @param applicationDeployment
+     */
+    public static void startExperiment(KruizeObject kruizeObject,
```
Comment (dinogun) [inline src/main/java/com/autotune/analyzer/Experimentator.java:45]: Is this class going to be phased out?

File: src/main/java/com/autotune/common/k8sObjects/ExperimentResultData.java (hunk 1 above)
Comment (msvinaykumar) [reply src/main/java/com/autotune/common/k8sObjects/ExperimentResultData.java]: Agreed moved to different directory.

File: src/main/java/com/autotune/service/InitiateListener.java [hunk 3]
```diff
@@ -51,25 +62,90 @@ public void contextInitialized(ServletContextEvent sce) {
...
-        AutotuneQueue<ExperimentTrial> emQueue = new AutotuneQueue<>(20000);
-        /**
-         * Thread pool executor declaration for Experiment Manager
-         */
         AutotuneExecutor EMExecutor = new AutotuneExecutor(ParallelEngineConfigs.EM_CORE_POOL_SIZE,
                 ParallelEngineConfigs.EM_MAX_POOL_SIZE,
                 ParallelEngineConfigs.EM_CORE_POOL_KEEPALIVETIME_IN_SECS,
                 TimeUnit.SECONDS,
-                emQueue,
+                new AutotuneQueue<>(20000),
```
[the remark this answers is not in the document]
Comment (msvinaykumar) [reply src/main/java/com/autotune/service/InitiateListener.java]: done

File: src/main/java/com/autotune/analyzer/Experimentator.java (hunk 2 above)
Comment (msvinaykumar) [reply src/main/java/com/autotune/analyzer/Experimentator.java]: yes.

File: src/main/java/com/autotune/analyzer/deployment/KruizeDeployment.java [hunk 4]
```diff
@@ -161,7 +161,7 @@ public static void processKruizeObject(KruizeObject kruizeObject) {
                 List<KruizeObject> kruizeObjectList = new ArrayList<>();
                 kruizeObjectList.add(kruizeObject);
                 ExperimentInitiator experimentInitiator = new ExperimentInitiator();
-                ActivityResultData activityResultData = experimentInitiator.validateAndAdd(autotuneObjectMap, kruizeObjectList);
+                ActivityResultData activityResultData = experimentInitiator.validateAndAddNewExperiments(autotuneObjectMap, kruizeObjectList);
                 if (!activityResultData.isSuccess()) {
```
Comment (dinogun) [inline src/main/java/com/autotune/analyzer/deployment/KruizeDeployment.java:165]: Can we unify the logging in both of the cases of starting and experiment (k8s native and API). Right now it appears the logging happens in one and not in the other.

File: src/main/java/com/autotune/analyzer/services/CreateExperiment.java [hunk 5]
```diff
@@ -73,7 +73,7 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         try {
             String inputData = request.getReader().lines().collect(Collectors.joining());
             List<KruizeObject> kruizeExpList = Arrays.asList(new Gson().fromJson(inputData, KruizeObject[].class));
-            ActivityResultData experimentInitiator = new ExperimentInitiator().validateAndAdd(mainKruizeExperimentMap, kruizeExpList);
+            ActivityResultData experimentInitiator = new ExperimentInitiator().validateAndAddNewExperiments(mainKruizeExperimentMap, kruizeExpList);
```
Comment (dinogun) [inline src/main/java/com/autotune/analyzer/services/CreateExperiment.java:76]: See the above message about unified logging

File: src/main/java/com/autotune/analyzer/services/UpdateResults.java [hunk 6]
```diff
@@ -61,41 +62,28 @@ public void init(ServletConfig config) throws ServletException {
...
-                        results = new ArrayList<>();
-                    else
-                        results = exp.getResults();
-                    results.add(jsonObject);
-                    exp.setResults(results);
-                }
+            List<ExperimentResultData> experimentResultDataList = Arrays.asList(new Gson().fromJson(inputData, ExperimentResultData[].class));
+            ActivityResultData experimentInitiator = new ExperimentInitiator().validateAndUpdateResults(mainKruizeExperimentMap, experimentResultDataList);
+            if (experimentInitiator.isSuccess()) {
+                sendSuccessResponse(response, "Results added successfully");
```
Comment (dinogun) [inline src/main/java/com/autotune/analyzer/services/UpdateResults.java:68]: Logging should happen inside `validateAndUpdateResults`, that way it is common in the case of both k8s native and REST API invocation

**Comment by chandrams** [comment]
@msvinaykumar - Can you please update the API doc with the latest updateResults json.

**Review by dinogun — APPROVED** [review]
LGTM!

**Pull request merged**